core/storage/leveldb/leveldb_util.hpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_LEVELDB_UTIL_HPP
+#define KAGOME_LEVELDB_UTIL_HPP
+
+#include <leveldb/status.h>
+#include <gsl/span>
+#include <outcome/outcome.hpp>
+
+#include "common/buffer.hpp"
+#include "common/logger.hpp"
+#include "storage/leveldb/leveldb_error.hpp"
+
+namespace kagome::storage {
+
+  template <typename T>
+  inline outcome::result<T> error_as_result(const leveldb::Status &s) {
+    if (s.IsNotFound()) {
+      return LevelDBError ::kNotFound;
+    }
+
+    if (s.IsIOError()) {
+      return LevelDBError::kIOError;
+    }
+
+    if (s.IsInvalidArgument()) {
+      return LevelDBError::kInvalidArgument;
+    }
+
+    if (s.IsCorruption()) {
+      return LevelDBError::kCorruption;
+    }
+
+    if (s.IsNotSupportedError()) {
+      return LevelDBError::kNotSupported;
+    }
+
+    return LevelDBError::kUnknown;
+  }
+
+  template <typename T>
+  inline outcome::result<T> error_as_result(const leveldb::Status &s,
+                                            const common::Logger &logger) {
+    logger->error(s.ToString());
+    return error_as_result<T>(s);
+  }
+
+  inline leveldb::Slice make_slice(const common::Buffer &buf) {
```

Comment:
why not to move implementations of these functions to cpp?

Reply:
There is no point. This method is very simple (just a shortcut to cast one type to another), so it can be inline.